### PR TITLE
Add rounded-corner-rules configuration option

### PR DESCRIPTION
**Pull request by alanpq**
As mentioned in #874, this PR adds a `rounded-corner-rules` config entry, to allow the user to specify windows that should have a different corner-radius to the default.

The name of the entry might want to be renamed, so I haven't added it to the man page yet.

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -508,6 +508,38 @@ bool parse_geometry(session_t *ps, const char *src, region_t *dest) {
...
+	if (radius < 0 || radius > INT_MAX) {
+		log_error("Invalid corner radius: %s", src);
+		return false;
+	}
+	
+	// Skip over spaces
+	while (*endptr && isspace((unsigned char)*endptr))
+		++endptr;
+	if (':' != *endptr) {
+		log_error("Opacity terminator not found: %s", src);
```
Comment (yshui) [inline src/config.c]: You mean "radius"? BTW, I am not really a fan of code duplication, can this share code with parse_rule_opacity?

**Comment by yshui** [comment]
Thanks for working on this! This looks good in general, just one nit.

File: src/config.c (hunk 1 above)
Comment (alanpq) [reply src/config.c]: Yeah I've merged them into one function now, the only duplication left is having to pass the same min/max params whenever you call it, but still much better IMO

Comment (yshui) [reply src/config.c]: Yes, thanks.

**Comment by yshui** [comment]
@alanpq thanks

**Pull request closed without merging**